### PR TITLE
validate label value

### PR DESCRIPTION
**Pull request by replay**
This adds another check to the Distributor's `validateLabels()` function which validates that received label values only contain utf-8 characters. I also added a test case which sends a label value in Chinese, to ensure that non-english character sets still work as expected.

[The Prometheus spec says](https://prometheus.io/docs/concepts/data_model/#metric-names-and-labels):
> Label values may contain any Unicode characters.

This also extends the scope of the `mimirpb.WriteRequest.SkipLabelNameValidation`, the flag makes the Distributor also skip the new label value validation which this PR adds. To make the name remain accurate, it renames it from `SkipLabelNameValidation` to `SkipLabelValidation`.

### CODE REVIEW

**Review by flxbk — APPROVED** [review]
Thanks, I only had one minor question.

File: pkg/distributor/validate.go [hunk 1]
```diff
@@ -400,6 +406,9 @@ func validateLabels(m *sampleValidationMetrics, cfg labelValidationConfig, userI
 		} else if len(l.Value) > maxLabelValueLength {
 			m.labelValueTooLong.WithLabelValues(userID, group).Inc()
 			return fmt.Errorf(labelValueTooLongMsgFormat, l.Name, l.Value, mimirpb.FromLabelAdaptersToString(ls))
+		} else if !model.LabelValue(l.Value).IsValid() {
```
[the remark this answers is not in the document]
Comment (replay) [reply pkg/distributor/validate.go:409]: discussed via DM

[the remark this answers is not in the document]
Comment (replay) [reply pkg/distributor/validate.go]: I updated this PR to make the flag also skip the new label value validation.

**Review by tacole02 — APPROVED** [review]
Looks good! Thanks for updating :)

**Comment by narqo** [comment]
The CHANGELOG has just been cut to prepare for the next release. Please rebase `main` and eventually move the CHANGELOG entry added / updated in this PR to the top of the CHANGELOG.md document. Thanks!

**Comment by replay** [comment]
> The CHANGELOG has just been cut to prepare for the next release. Please rebase main and eventually move the CHANGELOG entry added / updated in this PR to the top of the CHANGELOG.md document. Thanks!

Rebasing

**Comment by replay** [comment]
we can't merge this as it is, because it renames this HTTP header without backwards compatibility:

https://github.com/grafana/mimir/pull/9185/files#diff-8a770bd239c72b7de7bc51e232145f906f94d15727124a15fbe0d46c2d98def7R47

**Comment by replay** [comment]
I don't think it's necessary to go through the hassle of renaming the HTTP header `X-Mimir-SkipLabelNameValidation` to `X-Mimir-SkipLabelValidation` because label name validation is a subset of label name validation, so the expression `SkipLabelNameValidation` doesn't become incorrect just because the header disables the  validation of the label name and label value.

**Pull request merged**